read-fonts/src/tables/varc.rs

```diff
@@ -1,17 +1,139 @@
 //! the [VARC (Variable Composite/Component)](https://github.com/harfbuzz/boring-expansion-spec/blob/main/VARC.md) table
 
-pub use super::layout::CoverageTable;
+pub use super::layout::{Condition, CoverageTable};
 
 include!("../../generated/generated_varc.rs");
 
 #[cfg(test)]
 mod tests {
     use crate::{FontRef, TableProvider};
 
+    use super::{Condition, Varc};
+
+    fn read_conditions<'a>(table: &'a Varc) -> Vec<Condition<'a>> {
+        table
+            .condition_list()
+            .expect("A condition list")
+            .conditions()
+            .iter()
+            .enumerate()
+            .map(|(i, c)| c.unwrap_or_else(|e| panic!("condition {i} {e}")))
+            .collect()
+    }
+
     #[test]
     fn read_cjk_0x6868() {
         let font = FontRef::new(font_test_data::varc::CJK_6868).unwrap();
         let table = font.varc().unwrap();
         table.coverage().unwrap(); // should have coverage
     }
+
+    #[test]
+    fn identify_all_conditional_types() {
+        let font = FontRef::new(font_test_data::varc::CONDITIONALS).unwrap();
+        let table = font.varc().unwrap();
+
+        // We should have all 5 condition types in order
+        assert_eq!(
+            (1..=5).collect::<Vec<_>>(),
```

Comment:
iirc you can compare slices/arrays with vecs (there are `PartialEq` impls) so I think you can write, `assert_eq!([1, 2, 3, 4, 5], blah.collect::<Vec<_>>())`

Reply:
I'm getting ```can't compare `[{integer}; 5]` with `Vec<i32>``` so I will leave it for the time being

Reply:
ah okay the issue there might be that you need to be explicit about the integer type, like `[1i32, 2, 3, 4, 5]`